ADIOS 2.5.0: SSC also uses JSON

I am currently seeing a spack build issue with ADIOS 2.5.0 on CI of the form:

```
from /tmp/travis/spack-stage/adios2-2.5.0-tk6drsnu

            qt3zexkihltbumzfzi2j4a6b/spack-src/source/adios2/core/IO.cpp:42:

  >> 679    /tmp/travis/spack-stage/adios2-2.5.0-tk6drsnuqt3zexkihltbumzfzi2j4a

            6b/spack-src/source/adios2/toolkit/format/dataman/DataManSerializer

            .h:22:29: fatal error: nlohmann/json.hpp: No such file or directory

     680     #include <nlohmann/json.hpp>
```
https://travis-ci.org/openPMD/openPMD-api/jobs/598258587

The problem is likely in CMake blocks such as
https://github.com/ornladios/ADIOS2/blob/v2.5.0/source/adios2/CMakeLists.txt#L161-L170
that also include `toolkit/format/dataman/DataManSerializer.cpp|tcc` but do not link `nlohmann_json` as in Dataman, etc.

Spack spec:
`adios2@2.5.0%gcc@4.9.4+blosc build_type=Release +bzip2~dataman~endian_reverse~fortran~hdf5~mpi+pic+png~python+shared+ssc+sst+sz~zfp arch=linux-ubuntu14.04-haswell`

--- a/source/adios2/CMakeLists.txt
+++ b/source/adios2/CMakeLists.txt
@@ -141,7 +141,7 @@ if (ADIOS2_HAVE_DataMan)
     target_link_libraries(adios2 PRIVATE nlohmann_json)
 endif()
 
-if(ADIOS2_HAVE_DataMan OR ADIOS2_HAVE_WDM OR ADIOS2_HAVE_Table)
+if(ADIOS2_HAVE_DataMan OR ADIOS2_HAVE_WDM OR ADIOS2_HAVE_Table OR ADIOS2_HAVE_SSC)
     target_sources(adios2 PRIVATE
        	toolkit/format/dataman/DataManSerializer.cpp
         toolkit/format/dataman/DataManSerializer.tcc
@@ -162,8 +162,6 @@ endif()
 
 if(ADIOS2_HAVE_SSC)
     target_sources(adios2 PRIVATE
-        toolkit/format/dataman/DataManSerializer.cpp
-        toolkit/format/dataman/DataManSerializer.tcc
         engine/ssc/SscReader.cpp
         engine/ssc/SscReader.tcc
         engine/ssc/SscWriter.cpp

--- a/source/adios2/CMakeLists.txt
+++ b/source/adios2/CMakeLists.txt
@@ -143,9 +143,9 @@ endif()
 
 if(ADIOS2_HAVE_DataMan OR ADIOS2_HAVE_WDM OR ADIOS2_HAVE_Table OR ADIOS2_HAVE_SSC)
     target_sources(adios2 PRIVATE
-       	toolkit/format/dataman/DataManSerializer.cpp
+        toolkit/format/dataman/DataManSerializer.cpp
         toolkit/format/dataman/DataManSerializer.tcc
-        )
+    )
     target_link_libraries(adios2 PRIVATE nlohmann_json)
 endif()
 
@@ -166,14 +166,14 @@ if(ADIOS2_HAVE_SSC)
         engine/ssc/SscReader.tcc
         engine/ssc/SscWriter.cpp
         engine/ssc/SscWriter.tcc
-        )
+    )
 endif()
 
 if(ADIOS2_HAVE_Table)
     target_sources(adios2 PRIVATE
         engine/table/TableWriter.cpp
         engine/table/TableWriter.tcc
-        )
+    )
 endif()
 
 if(ADIOS2_HAVE_SST)

